agent/org.linkedin.glu.agent-server-upgrade/src/cmdline/resources/bin/agentctl.sh

```diff
@@ -202,14 +204,19 @@ get_pid()
     PID=`cat $PID_FILE`
     PS_STAT=`ps -p $PID -o'user,pid=' | tail -1 | awk '{print $2}'`
 
-    case "$PS_STAT" in
-      $PID  ) echo $PID
-              ;;
-      *     ) echo 0
-              ;;
-    esac
+    if [[ "$PS_STAT" -eq "$PID" ]]; then
+      CMD=$(ps -p $PID -o 'comm=')
```

Comment:
basename is part of GNU coreutils on Linux and part of BSD general commands on Mac.  The mac man page says basename is part of the POSIX.2 standard so I'm pretty sure other Unix variants will have it as well.

We could check the command line arguments like you suggest but the test is more complicated in shell and I don't really think it is necessary.  Since I'm removing the PID_FILE on clean shutdowns it is less likely that we'll have a stale PID_FILE.  My systems have very few java processes so the odds of having a stale PID_FILE and having the PID wrap and get used by a different java process are very small.  Perhaps others run lots of java processes, but I seem to be the first person who has hit (or at least reported/fixed) this issue in the first place which means it was already not that common even before my change.  Maybe LinkedIn doesn't reboot or doesn't start glu-agent at startup.